ochk/resource_custom_service_test.go

```diff
@@ -0,0 +1,148 @@
+package ochk
+
+import (
+	"context"
+	"fmt"
+	"github.com/chmurakrajowa/terraform-provider-ochk/ochk/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"testing"
+)
+
+type CustomServiceTestData struct {
+	ResourceName string
+	DisplayName  string
+	Ports        []CustomServicePortTestData
+}
+
+type CustomServicePortTestData struct {
+	Protocol    string
+	Source      []string
+	Destination []string
+}
+
+func (c *CustomServiceTestData) ToString() string {
+	return executeTemplateToString(`
+resource "ochk_custom_service" "{{ .ResourceName}}" {
+  display_name = "{{ .DisplayName}}"
+
+  {{range $port := .Ports}}
+  ports {
+    protocol = "{{ $port.Protocol }}"
+    source = {{ StringsToTFList $port.Source }}
+    destination = {{ StringsToTFList $port.Destination }}
+  }
+  {{end}}
+}
+`, c)
+}
+
+func (c *CustomServiceTestData) FullResourceName() string {
+	return "ochk_custom_service." + c.ResourceName
+}
+
+func TestAccCustomServiceResource_create(t *testing.T) {
```

Comment:
Tak się zastanawiam czy nie warto zrobić testów, które używają tego CustomService w tworzeniu innych zasobów. Coś mi kiedyś mówił Adrian, że można je wykorzystywać zamiast DefaultService, ciekawe czy to działa. Co o tym myślisz?

Reply:
Tak tak, taki jest plan. Jeszcze dzisiaj to dorobię

Reply:
Będzie jednak przy okazji innego PRa w ramach zmian w FW.